internal/app/visit/mongodb_repo.go

```diff
@@ -0,0 +1,72 @@
+package visit
+
+import (
+	"context"
+
+	"github.com/TheFlies/ofriends/internal/app/types"
+	"github.com/TheFlies/ofriends/internal/pkg/uuid"
+	mgo "github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	"github.com/pkg/errors"
+)
+
+// MongoRepository is MongoDB implementation of repository
+type MongoRepository struct {
+	session *mgo.Session
+}
+
+// NewMongoRepository return new MongoDB repository
+func NewMongoRepository(s *mgo.Session) *MongoRepository {
+	return &MongoRepository{
+		session: s,
+	}
+}
+
+// FindByID return friend base on given id
+func (r *MongoRepository) FindByID(ctx context.Context, id string) (*types.Visit, error) {
+	s := r.session.Clone()
+	defer s.Close()
+	var visit *types.Visit
+	if err := r.collection(s).Find(bson.M{"_id": id}).One(&visit); err != nil {
+		return nil, errors.Wrap(err, "failed to find the given visit from database")
+	}
+	return visit, nil
+}
+
+// FindAll return all visits
+func (r *MongoRepository) FindAll(ctx context.Context) ([]types.Visit, error) {
+	s := r.session.Clone()
+	defer s.Close()
+	var visit []types.Visit
+	if err := r.collection(s).Find(bson.M{}).All(&visit); err != nil {
+		return nil, errors.Wrap(err, "failed to fetch all visit from database")
+	}
+	
+	return visit, nil
+}
+
+// Create a visit
+func (r *MongoRepository) Create(ctx context.Context, visit types.Visit) error {
```

Comment:
response should be (string, error) where string is the id of the record that you have inserted into db